File: C++/intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        
        int sz1 = nums1.size();
        int sz2 = nums2.size();
        
        vector<int> result;
        
        for (int i = 0, j = 0; i < sz1 && j < sz2; ) {
            if (nums1[i] == nums2[j]) {
                result.push_back(nums1[i]);
                i++;
                j++;
            } else if (nums1[i] < nums2[j]) {
                i++;
            } else {
                j++;
            }
        }
        
        return result;
    }
};

File: Go/intersection-of-two-arrays-ii.go
func intersect(nums1 []int, nums2 []int) []int {
	count := [1001]int{}
	for _, n := range nums1 {
		count[n]++
	}
    res := []int{}
	for _, n := range nums2 {
		if count[n] > 0 {
			res = append(res, n)
			count[n]--
		}
	}
	return res
}